### PR TITLE
[#8] [#10] [Chore] [Backend] Setup API Service & As a user, I can login 

### LINKED ISSUE #8: [Backend] As a user, I can login
## Why

* Request Sign In API to sign user in using user's input: email and password

## Acceptance Criteria

* Create Repository and UseCase
* Call API [Sign In](https://documenter.getpostman.com/view/11835486/UVsPNPfU#deb6f64d-60a4-434b-9c83-215ad8a07568)
* Write Unit Tests

### PR DESCRIPTION
**Pull request by thiennguyen0196**
- Close #8 #10 

## What happened 👀

- Setup API service
- Setup repository
- Setup dependencies injection
- Create response and model data
- Create login use case
- Add unit test

## Insight 📝

- Run `flutter pub get` and `flutter pub run build_runner build ` to get dependencies and generate necessary files.

- The github secret and the local repository should include these variables as valid values from the POSTMAN environment
  - REST_API_ENDPOINT
  - CLIENT_ID
  - CLIENT_SECRET

- The `Storage` and `AppInterceptor` will be finished later in the integration task, right now it is in the initial phase (not completed 100%)

## Proof Of Work 📹

All the unit tests passed

Screenshot 2023-04-14 at 3 17 38 PM

### CODE REVIEW

File: .env
Comment (nmint8m) [inline .env:1]: Please don't commit the file `.env`. As we stored those sensitive data in the secrets already. 

https://github.com/nimblehq/flutter-ic-david-bruce/blob/e741df3ec2b43d640ab60250bfd62be0173f2b36/.github/workflows/deploy_ios_staging.yml#L40

File: .env.staging
Comment (nmint8m) [inline .env.staging:1]: Please don't commit the file `.env.staging`. As we stored those sensitive data in the secrets already. 

https://github.com/nimblehq/flutter-ic-david-bruce/blob/e741df3ec2b43d640ab60250bfd62be0173f2b36/.github/workflows/deploy_ios_staging.yml#L40

File: lib/model/response/login_response.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  final String tokenType;
+
+  @JsonKey(name: 'expires_in')
+  final double expiresIn;
+
+  @JsonKey(name: 'refresh_token')
+  final String refreshToken;
+
+  @JsonKey(name: 'created_at')
+  final double createdAt;
```
Comment (nmint8m) [inline lib/model/response/login_response.dart:28]: Please consider removing `@JsonKey` if not neccessary.

**Review by nmint8m — COMMENTED** [review]
Please take a look

File: lib/model/request/login_request.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  final String clientSecret;
+
+  @JsonKey(name: 'grant_type')
+  final String grantType;
+
+  @JsonKey(name: 'email')
+  final String email;
+
+  @JsonKey(name: 'password')
+  final String password;
```
[the remark this answers is not in the document]
Comment (thiennguyen0196) [reply lib/model/request/login_request.dart:22]: It's a habit from Android side 😄 you got the point here to not use the `@JsonKey` anymore 👍 Many thanks @nmint8m  👏

**Review by nmint8m — APPROVED** [review]
Look good to me.

**Review by ducbm051291 — APPROVED** [review]

**Pull request merged**